cmd/kube-controller-manager/app/options/options.go

```diff
@@ -225,38 +225,39 @@ func NewDefaultComponentConfig(insecurePort int32) (componentconfig.KubeControll
 	return internal, nil
 }
 
-// AddFlags adds flags for a specific KubeControllerManagerOptions to the specified FlagSet
-func (s *KubeControllerManagerOptions) AddFlags(fs *pflag.FlagSet, allControllers []string, disabledByDefaultControllers []string) {
-	s.CloudProvider.AddFlags(fs)
-	s.Debugging.AddFlags(fs)
-	s.GenericComponent.AddFlags(fs)
-	s.KubeCloudShared.AddFlags(fs)
-	s.ServiceController.AddFlags(fs)
-
-	s.SecureServing.AddFlags(fs)
-	s.InsecureServing.AddFlags(fs)
-	s.Authentication.AddFlags(fs)
-	s.Authorization.AddFlags(fs)
-
-	s.AttachDetachController.AddFlags(fs)
-	s.CSRSigningController.AddFlags(fs)
-	s.DeploymentController.AddFlags(fs)
-	s.DaemonSetController.AddFlags(fs)
-	s.DeprecatedFlags.AddFlags(fs)
-	s.EndPointController.AddFlags(fs)
-	s.GarbageCollectorController.AddFlags(fs)
-	s.HPAController.AddFlags(fs)
-	s.JobController.AddFlags(fs)
-	s.NamespaceController.AddFlags(fs)
-	s.NodeIpamController.AddFlags(fs)
-	s.NodeLifecycleController.AddFlags(fs)
-	s.PersistentVolumeBinderController.AddFlags(fs)
-	s.PodGCController.AddFlags(fs)
-	s.ReplicaSetController.AddFlags(fs)
-	s.ReplicationController.AddFlags(fs)
-	s.ResourceQuotaController.AddFlags(fs)
-	s.SAController.AddFlags(fs)
-
+// Flags returns flags for a specific APIServer by section name
+func (s *KubeControllerManagerOptions) Flags(allControllers []string, disabledByDefaultControllers []string) (fss apiserverflag.NamedFlagSets) {
+	s.CloudProvider.AddFlags(fss.FlagSet("cloud provider"))
+	s.Debugging.AddFlags(fss.FlagSet("debugging"))
+	s.GenericComponent.AddFlags(fss.FlagSet("generic"))
+	s.KubeCloudShared.AddFlags(fss.FlagSet("kube-cloud shared"))
+	s.ServiceController.AddFlags(fss.FlagSet("service controller"))
+
+	s.SecureServing.AddFlags(fss.FlagSet("secure serving"))
+	s.InsecureServing.AddFlags(fss.FlagSet("insecure serving"))
+	s.Authentication.AddFlags(fss.FlagSet("authentication"))
+	s.Authorization.AddFlags(fss.FlagSet("authorization"))
+
+	s.AttachDetachController.AddFlags(fss.FlagSet("attachdetach controller"))
+	s.CSRSigningController.AddFlags(fss.FlagSet("csrsigning controller"))
+	s.DeploymentController.AddFlags(fss.FlagSet("deployment controller"))
+	s.DaemonSetController.AddFlags(fss.FlagSet("daemonset controller"))
+	s.DeprecatedFlags.AddFlags(fss.FlagSet("deprecated"))
+	s.EndPointController.AddFlags(fss.FlagSet("endpoint controller"))
+	s.GarbageCollectorController.AddFlags(fss.FlagSet("garbagecollector controller"))
+	s.HPAController.AddFlags(fss.FlagSet("horizontalpodautoscaling controller"))
```

Comment:
maybe CamelCase those strings?

Reply:
fixed, and add judgement for one option section has no flags